differs/size_diff.go

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright 2018 Google, Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package differs
+
+import (
+	"strconv"
+
+	pkgutil "github.com/GoogleContainerTools/container-diff/pkg/util"
+	"github.com/GoogleContainerTools/container-diff/util"
+)
+
+type SizeAnalyzer struct {
+}
+
+func (a SizeAnalyzer) Name() string {
+	return "SizeAnalyzer"
+}
+
+// SizeDiff diffs two images and compares their size
+func (a SizeAnalyzer) Diff(image1, image2 pkgutil.Image) (util.Result, error) {
+	diff := []util.EntryDiff{
+		{
+			Name:  "*",
```

Comment:
is there a reason you used `*` here? should this just be the empty string?

Reply:
The reason there is an `*` here is because it indicates "all layers" of the image. I was trying to use templates efficiently and just created two for the four different cases.

The four cases are:
1. `analyze <image> --type=size`
2. `analyze <image> --type=sizelayer`
3. `diff <image1> <image2> --type=size`
4. `diff <image1> <image2> --type=sizelayer`

Rather than create four separate templates I found a way to create two, one for analysis and one for diff. Both templates have `LAYER` as the first column and I reuse them for the image size comparison by displaying an `*`. Here is an example:

```bash
$ out/container-diff analyze gcr.io/gcp-runtimes/diff-layer-base --type=size

-----Size-----

Analysis for gcr.io/gcp-runtimes/diff-layer-base:
LAYER        SIZE
*            19B
```

When analyzing/diffing the image layers the layer number is shown as in this example:

```bash
$ out/container-diff analyze gcr.io/gcp-runtimes/diff-layer-base --type=sizelayer

-----SizeLayer-----

Analysis for gcr.io/gcp-runtimes/diff-layer-base:
LAYER        SIZE
0            6B
1            7B
2            6B
```

If you would prefer to have the column read `IMAGE` for the image size comparison cases I can create two further templates. Let me know what you think.

Reply:
Ah I see what you did. Being totally honest, I'm not a huge fan of reusing `LAYER` for the image diff, I think it's a little confusing. If you could create another template for this case that would be great.